test/cpp/microbenchmarks/bm_filters.cc

```diff
@@ -0,0 +1,200 @@
+/*
+ *
+ * Copyright 2018 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <benchmark/benchmark.h>
+#include <string.h>
+#include <sstream>
+
+#include <grpc/grpc.h>
+#include <grpc/support/alloc.h>
+#include <grpc/support/string_util.h>
+#include <grpcpp/channel.h>
+#include <grpcpp/support/channel_arguments.h>
+
+#include "src/core/ext/filters/client_channel/client_channel.h"
+#include "src/core/ext/filters/deadline/deadline_filter.h"
+#include "src/core/ext/filters/http/client/http_client_filter.h"
+#include "src/core/ext/filters/http/message_compress/message_compress_filter.h"
+#include "src/core/ext/filters/http/server/http_server_filter.h"
+#include "src/core/ext/filters/load_reporting/server_load_reporting_filter.h"
+#include "src/core/ext/filters/message_size/message_size_filter.h"
+#include "src/core/lib/gprpp/manual_constructor.h"
+
+#include "src/cpp/client/create_channel_internal.h"
+#include "src/proto/grpc/testing/echo.grpc.pb.h"
+#include "test/cpp/microbenchmarks/filter_helpers.h"
+#include "test/cpp/util/test_config.h"
+
+auto& force_library_initialization = Library::get();
+
+// Test a filter's call stack init in isolation. FilterBM, in conjunction with
+// FilterFixture, specifies the filter under test (use the FilterBM<> template
+// to specify this).
+// Note that there is some other work being done within the benchmarking loop,
+// so the result of this microbenchmark is a little bloated.
+template <class FilterBM>
+static void BM_CallStackInit(benchmark::State& state) {
+  // Setup for benchmark
+  FilterBM bm_setup;
+  struct DataForFilterBM data;
+  bm_setup.Setup(&data);
+
+  // Run the benchmark
+  grpc_call_final_info final_info;
+  while (state.KeepRunning()) {
+    GPR_TIMER_SCOPE("BenchmarkCycle", 0);
```

Comment:
Is this needed for any reason? It will be a no-op if we are not running a `counters` build, right?

Reply:
Hahaha, I went back and time and found the commit, then found the PR that introduced it. I reviewed it :) #10469

They provide no value for this benchmark, since we never run with timers enabled.

Reply:
haha ok good digging :) I'll remove them from these benchmarks.